### PR TITLE
Foulab Events - Jan 2025

### PR DESCRIPTION
**Pull request by v2thegreat**
This pull request adds a new article to the `content/news/events-january-2025.md` file, detailing upcoming events at Foulab in January 2025. The article includes event descriptions, dates, and times, along with some engaging commentary and invitations to participate.

Key changes include:

* **Event Listings and Descriptions**:
  * Added detailed descriptions for several events, including the Wednesday Mesh Evening, Workshop Reorganization Planning, War Gaming, FOULAB Monthly Meeting, Project Night, and Killed by Dice Presents: Three Sixteen V.

* **Formatting and Structure**:
  * Structured the article with headlines, titles, and proper formatting to ensure readability and engagement.

* **Community Engagement**:
  * Encouraged community participation and highlighted the collaborative nature of the events, with specific mentions of hosts and collaborators.

### CODE REVIEW

**Comment by cpatulea** [comment]
Happened to notice this, thanks for working on it, looks like a good first draft! Let us know when it's ready for review.

File: content/news/events-january-2025.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+Everyone is hosting, and **kryma** invites you to host as well. 
+
+The network must grow! The network must grow! The network must grow! The network must grow! The network must grow! The network must grow! The network mu- (the author has been disconnected from the network)
+
+<br/>
+
+
+### Thursday, January 9th, 8:00PM to 9:30PM: Workshop Reorganization Planning!
+
+The foulab workshop area needs some love! We have a lot of tools, both traditional (hand tools, drill press, bandsaw, air compressor, welder, etc) and modern (3D printer, CNC mills, laser cutter) but not a lot of good places to put them.
```
Comment (cpatulea) [inline content/news/events-january-2025.md]: I suggest stylize as 'Foulab' (capital F), consistent with the style in most places.

File: content/news/events-january-2025.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+We do, however, have an excess of storage racking left over from the museum expansion, so let's solve two problems at once and reorganize the workshop! 
+
+This is the initial planning meeting, to measure out the tools we have, propose racking layouts, and prepare for the work that lies ahead. Anyone interested in helping out is invited to attend! 
+
+<br/>
+
+
+### Open Gaming Night - Tuesday January 21st @ # TIMINGS ARE TO BE CONFIRMED
+
+Starting in January, we've opened up the monthly Warhammer 40k gaming night to include other games. This month, der_moderne_man will bring Steve Jackson's fantasy dueling games Melee & Wizard. Others are encouraged to bring games they want to play. Warhammer? Magic the Gathering? Risk? Mouseguard: Swords & Strongholds?
```
Comment (cpatulea) [inline content/news/events-january-2025.md]: der_moderne_man: nicknames usually stylized using bold (**).

File: content/news/events-january-2025.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+Starting in January, we've opened up the monthly Warhammer 40k gaming night to include other games. This month, der_moderne_man will bring Steve Jackson's fantasy dueling games Melee & Wizard. Others are encouraged to bring games they want to play. Warhammer? Magic the Gathering? Risk? Mouseguard: Swords & Strongholds? 
+
+The more the merrier, bring yourself, bring a game if you have one, we'll match you up with other players! 
+
+<br/>
+
+
+### FOULAB Monthly Meeting - Thursday January 23rd @ 20:00 - 21:30
+
+The first meeting of the year! Make a new year's resolution to come to this meeting and make your voice heard on things that mattermost.
```
Comment (cpatulea) [inline content/news/events-january-2025.md]: new year's: usually capitalized (https://en.wikipedia.org/wiki/New_Year's_resolution)

File: content/news/events-january-2025.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
++++
+
+{{< centered-image src="events-jan-2025.gif" width="384">}}
+
+
+<center><i>An early 3D model of the workshop by **v2thegreat**</i></center>
+
+<br/>
+
+Do you feel it? We're a quarter of the way through the 21st century! Feel old yet? Surpress those feelings with some good old-fashioned hacking and making! Here's what's coming up in January:
```
Comment (cpatulea) [inline content/news/events-january-2025.md]: "Surpress": intentional or typo?

**Review by cpatulea — CHANGES_REQUESTED** [review]

File: content/news/events-january-2025.md (hunk 4 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md]: No I'm just dumb lol

**Comment by v2thegreat** [comment]
@cpatulea changes done!

File: content/news/events-january-2025.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+### Open Gaming Night - Tuesday January 21st @ # TIMINGS ARE TO BE CONFIRMED
+
+Starting in January, we've opened up the monthly Warhammer 40k gaming night to include other games. This month, **der_moderne_man** will bring Steve Jackson's fantasy dueling games Melee & Wizard. Others are encouraged to bring games they want to play. Warhammer? Magic the Gathering? Risk? Mouseguard: Swords & Strongholds? 
+
+The more the merrier, bring yourself, bring a game if you have one, we'll match you up with other players! 
+
+<br/>
+
+
+### Foulab Monthly Meeting - Thursday January 23rd @ 20:00 - 21:30
```
Comment (cpatulea) [inline content/news/events-january-2025.md]: Sorry, the 'FOULAB Monthly Meeting' event title is usually spelled in all caps (references: https://foulab.org/news/events-december-2024/, https://foulab.org/news/events-november-2024/), I don't know the initial rationale, but I think we should keep it that way, you had it right the first time.

**Review by cpatulea — CHANGES_REQUESTED** [review]

File: content/news/events-january-2025.md (hunk 5 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md]: I'll play devil's advocate to avoid a cargo cult here: are we sure we didn't have it as an early typo that we never realized until now? 

I'm fine either way, but it'd be funny if our branding were off literally because someone forever ago thought that **FOULAB** looked better than **Foulab**. 

Either way, here's a GH suggestions for what you think might be best

```suggestion
### FOULAB Monthly Meeting - Thursday January 23rd @ 20:00 - 21:30
```

Comment (cpatulea) [reply content/news/events-january-2025.md]: Fair enough. We'll need to wait for @kryma-foulab to get the history.

Comment (v2thegreat) [reply content/news/events-january-2025.md]: PS, I personally think FOULAB looks better than Foulab

Comment (kryma-foulab) [reply content/news/events-january-2025.md]: FOULAB is the way it's spelled in the header, but Foulab is more proper titlecase...  Let's stick with whatever it was before. There are good arguments for both but I recommend for consistency above either of those arguments - there'll always be time to switch things up.

File: static/media/image/news/events-jan-2025/events-jan-2025.gif
Comment (kryma-foulab) [inline static/media/image/news/events-jan-2025/events-jan-2025.gif:1]: wizardry :sparkles: 

Does it work rendered?

File: content/news/events-january-2025.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+Join the network! Explore mesh networking in all its forms. Discuss, research, test the unknown, and deploy peer-to-peer infrastructure!
+
+Everyone is hosting, and **kryma** invites you to host as well. 
+
+The network must grow! The network must grow! The network must grow! The network must grow! The network must grow! The network must grow! The network mu- (the author has been disconnected from the network)
+
+<br/>
+
+
+### Thursday, January 9th, 8:00PM to 9:30PM: Workshop Reorganization Planning!
```
Comment (kryma-foulab) [inline content/news/events-january-2025.md]: title format

File: content/news/events-january-2025.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+The Foulab workshop area needs some love! We have a lot of tools, both traditional (hand tools, drill press, bandsaw, air compressor, welder, etc) and modern (3D printer, CNC mills, laser cutter) but not a lot of good places to put them.
+
+We do, however, have an excess of storage racking left over from the museum expansion, so let's solve two problems at once and reorganize the workshop! 
+
+This is the initial planning meeting, to measure out the tools we have, propose racking layouts, and prepare for the work that lies ahead. Anyone interested in helping out is invited to attend! 
+
+<br/>
+
+
+### Open Gaming Night - Tuesday January 21st @ # TIMINGS ARE TO BE CONFIRMED
```
Comment (kryma-foulab) [inline content/news/events-january-2025.md]: TODO

File: content/news/events-january-2025.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+Starting in January, we've opened up the monthly Warhammer 40k gaming night to include other games. This month, **der_moderne_man** will bring Steve Jackson's fantasy dueling games Melee & Wizard. Others are encouraged to bring games they want to play. Warhammer? Magic the Gathering? Risk? Mouseguard: Swords & Strongholds? 
+
+The more the merrier, bring yourself, bring a game if you have one, we'll match you up with other players!
```
Comment (kryma-foulab) [inline content/news/events-january-2025.md]: "we'll match you up with other players" - could you rephrase? this is an unstaffed event really, more a hangout than anything else

File: static/media/image/news/events-jan-2025/events-jan-2025.gif
Comment (v2thegreat) [reply static/media/image/news/events-jan-2025/events-jan-2025.gif:1]: Yes. 

I tried it out on my end, and everything seems to work as expected. The image wouldn't load, but that was because the link was pointing towards the current server, where this hasn't been merged yet, so nbd

File: content/news/events-january-2025.md (hunk 6 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md:31]: What's the problem?

File: content/news/events-january-2025.md (hunk 8 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md]: I'll just remove the line.

```suggestion
The more the merrier! 
```

File: content/news/events-january-2025.md (hunk 7 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md]: Still no update from D, which is why there is no time provided

File: content/news/events-january-2025.md (hunk 6 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md]: good catch

```suggestion
### Workshop Reorganization Planning! - Thursday January 9th @ 20:00 - 21:30
```

File: content/news/events-january-2025.md (hunk 7 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md]: I'm planning on leaving it as is right now, then updating later. Is that ok with you?

File: generate_calendar.py
Comment (v2thegreat) [inline generate_calendar.py:1]: @kryma-foulab 

I was having a hard time following along with what the original script did, so if it's ok with you, I took a few minutes and cleaned it up a bit. I think it's more accessible now.

There's also a pip library called `ics` that's doing basically what you've done here much cleaner, but I can imagine you wanted to avoid additional dependencies for future people. If that requirement changes (or we add a good README) we should be able to integrate that too.

I added this in a separate commit, so I can drop it if you think this shouldn't be kept

File: content/news/events-january-2025.md (hunk 8 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md]: Oh, I just realized this was something that D wrote himself, which is why it was included

File: generate_calendar.py
Comment (v2thegreat) [reply generate_calendar.py:1]: PS - Found a bug: since the year is `now.year`, and this is for 2025, the outputs were for the wrong year. I added a patch for that

File: content/news/events-january-2025.md (hunk 7 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md]: Update: D responded, and times have been updated

File: content/news/events-january-2025.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+Project night is back! Did you work on something exciting over the holidays? Bring it to project night and show it off! 
+
+Let your creativity (and ego) shine! Practice your presentation skills, and get feedback from your peers. Talk about the great things you've done, and the challenges you've faced. 
+
+Beginners welcome! If you're new to the space, come by and see what we're all about. It's a great way to meet people and get involved in the community.
+
+<br/>
+
+
+### Killed by Dice Presents: Three Sixteen V - Saturday January 26th @ 16:00 - 21:00
```
Comment (str4w) [inline content/news/events-january-2025.md]: Saturday is Jan 25 - is this Saturday or Sunday?

File: static/ical/foulab.ics [hunk 10]
```diff
@@ -1296,5 +1296,67 @@ DESCRIPTION:Data Science in Python
 LOCATION:Foulab\, 999 Rue du Collège\, Montréal\, QC H4C 2S2\, Canada
 END:VEVENT
 
+X-EVENTMONTH:JAN2025
```
Comment (kryma-foulab) [inline static/ical/foulab.ics:1299]: What is this?

Comment (v2thegreat) [reply static/ical/foulab.ics:1299]: https://stackoverflow.com/a/43890138

Comment (v2thegreat) [reply static/ical/foulab.ics:1299]: I had a hard time finding where my months began

File: content/news/events-january-2025.md (hunk 9 above)
Comment (v2thegreat) [reply content/news/events-january-2025.md]: Good catch. I'll review quickly

Comment (v2thegreat) [reply content/news/events-january-2025.md]: Issue resolved and changes pushed. No need to update `ics` since that's generated on the date itself. 

ics had only datestamp updated when I checked

File: generate_calendar.py
Comment (cpatulea) [inline generate_calendar.py:1]: Would you mind fixing the filename in commit message 'generate_calendary.py'?

**Review by cpatulea — CHANGES_REQUESTED** [review]

File: generate_calendar.py
Comment (v2thegreat) [reply generate_calendar.py:1]: Done!

**Review by cpatulea — APPROVED** [review]

File: static/ical/foulab.ics (hunk 10 above)
Comment (cpatulea) [reply static/ical/foulab.ics:1299]: FYI https://github.com/arran4/golang-ical is not capable of parsing this, it fails with:

```
2025/01/09 23:25:58 Parse calendar error: malformed calendar; expected begin or end
```

https://icalendar.org/validator.html doesn't complain about this, so I guess the file is correct. There are no GitHub issues about this on golang-ical. So maybe it's a very niche but valid feature?

I will have to change to a different parser which is .. nrgh .. but I guess I have no choice.

Comment (cpatulea) [reply static/ical/foulab.ics:1299]: [edit] Sorry, https://github.com/apognu/gocal does work.
